Bonsai/SystemResourcesExtensionsSupport.cs

```diff
@@ -13,22 +14,30 @@ namespace Bonsai
     /// </remarks>
     internal static class SystemResourcesExtensionsSupport
     {
+        // Assembly.Load(byte[]) always results in a new assembly so we need to manually de-duplicate them
+        private static ConcurrentDictionary<string, Assembly> resolveCache = new();
```

Comment:
Do we really need the `ConcurrentDictionary` here? My understanding was that the `AssemblyResolve` event is never called concurrently, which is why I used a normal `Dictionary` in `ConfigurationHelper`.

I was trying to find a reference for this on the [`AssemblyResolve`](https://learn.microsoft.com/en-us/dotnet/api/system.appdomain.assemblyresolve?view=net-8.0) docs and in the [Resolve assembly loads guidance](https://learn.microsoft.com/en-us/dotnet/standard/assembly/resolve-loads) but could not find anything conclusive unfortunately, although I would indeed expect that if this is an issue someone would have flagged it. There is a [SO question](https://stackoverflow.com/questions/9180638/working-with-appdomain-assemblyresolve-event#9180843) that also uses a simple `IDictionary` and again no comments decrying its use, but surprised that I cannot find something definitive on this.

If it turns out to be the case we do need it, I would say we should probably change this in `ConfigurationHelper` as well, probably even as part of this same PR.

Reply:
Decided to adopt caching for all assembly resolution in #1904.

Reply:
(For the sake of posterity: I messaged Gonçalo on Teams to let him know `ConfigurationHelper` had this problem before I saw this and we talked about it there.)

> I am surprised by their choice of returning `null` in case the assembly is already in the process of being resolved, since it seems this might result in assembly resolution exceptions potentially being thrown from weird places at weird times.

In this case I think that behavior is very specific to the recursion issue they're working around. They actually intentionally want the inner resolves to fail.